### PR TITLE
[issue-730] Upgrade to flink 1.18.1

### PR DESCRIPTION
**Pull request by GaMalhot**
**Change log description**
1. Upgrade of Flink version to 1.18.1 on master branch
2. Changes w.r.t [FLINK_32376](https://github.com/apache/flink/pull/22840/commits/b14aae6c197d3b34021286b1d882af14930d4cb5) for Sink Init Context
3. Changes w.r.t [FLINK-31972](https://github.com/apache/flink/pull/22503/files) for fixing the ambiguous usage of `assertThat`

**Purpose of the change**
Fix: #730 

**What the code does**
Upgrade flinkVersion to 1.18.1 in gradle.properties

**How to verify it**
./gradlew clean build should pass

### CODE REVIEW

File: src/test/java/io/pravega/connectors/flink/PravegaCatalogITCase.java [hunk 1]
```diff
@@ -135,7 +135,7 @@ public void testCreateCatalogFromFactory() {
         assertThat(actualCatalog instanceof PravegaCatalog).isTrue();
         assertThat(((PravegaCatalog) actualCatalog).getName()).isEqualTo(CATALOG.getName());
         assertThat(((PravegaCatalog) actualCatalog).getDefaultDatabase()).isEqualTo(CATALOG.getDefaultDatabase());
-        assertThat(Whitebox.getInternalState(actualCatalog, "properties"))
+        org.assertj.core.api.Assertions.assertThat((Map) Whitebox.getInternalState(actualCatalog, "properties"))
```
Comment (crazyzhou) [inline src/test/java/io/pravega/connectors/flink/PravegaCatalogITCase.java:138]: We can use `assertThat` directly

File: src/test/java/io/pravega/connectors/flink/sink/PravegaSinkWriterITCase.java [hunk 2]
```diff
@@ -146,6 +148,21 @@ private static class SinkInitContext implements Sink.InitContext {
         SinkInitContext(
                 SinkWriterMetricGroup metricGroup) {
             this.metricGroup = metricGroup;
+                }
```
Comment (crazyzhou) [inline src/test/java/io/pravega/connectors/flink/sink/PravegaSinkWriterITCase.java:151]: Indentation not correct here.

**Review by crazyzhou — COMMENTED** [review]
Please add some reference Flink tickets of your code changes with your PR description, so that we can keep track, like FLINK-32376 and FLINK-31972

**Comment by GaMalhot** [comment]
> Please add some reference Flink tickets of your code changes with your PR description, so that we can keep track, like FLINK-32376 and FLINK-31972

Hi @crazyzhou , I have made changes and added the references in the description.

File: src/test/java/io/pravega/connectors/flink/sink/PravegaSinkWriterITCase.java (hunk 2 above)
Comment (GaMalhot) [reply src/test/java/io/pravega/connectors/flink/sink/PravegaSinkWriterITCase.java]: Hi Brian, The indentation has been fixed.

File: src/test/java/io/pravega/connectors/flink/PravegaCatalogITCase.java (hunk 1 above)
Comment (GaMalhot) [reply src/test/java/io/pravega/connectors/flink/PravegaCatalogITCase.java]: Hi Brian, the issue has been handled.

**Review by fyang86 — APPROVED** [review]
LGTM

**Review by ShwethaSNayak — APPROVED** [review]
LGTM

**Pull request merged**